packages/emitter-framework/src/python/components/enum-declaration/enum-declaration.tsx

```diff
@@ -0,0 +1,93 @@
+import { type Children, For, type Refkey } from "@alloy-js/core";
+import * as py from "@alloy-js/python";
+import type { Enum, EnumMember as TspEnumMember, Union } from "@typespec/compiler";
+import { useTsp } from "../../../core/context/tsp-context.js";
+import { reportDiagnostic } from "../../../lib.js";
+import { declarationRefkeys, efRefkey } from "../../utils/refkey.js";
+
+export interface EnumDeclarationProps extends Omit<py.BaseDeclarationProps, "name"> {
+  name?: string;
+  type: Union | Enum;
+}
+
+// Determine the appropriate enum type based on the member values
+function determineEnumType(
+  members: Array<[string, TspEnumMember]>,
+  type: Enum,
+): "IntFlag" | "Flag" | "IntEnum" | "StrEnum" | "Enum" {
+  const allInteger = members.every(([, member]) => {
+    const value = member.value ?? member.name;
+    return typeof value === "number" && Number.isInteger(value);
+  });
+
+  const allString = members.every(([, member]) => {
+    const value = member.value ?? member.name;
+    return typeof value === "string";
+  });
+
+  if (allInteger) {
+    return "IntEnum";
+  } else if (allString) {
+    return "StrEnum";
+  } else {
+    return "Enum";
+  }
+}
+
+export function EnumDeclaration(props: EnumDeclarationProps) {
+  const { $ } = useTsp();
+  let type: Enum;
+  if ($.union.is(props.type)) {
+    if (!$.union.isValidEnum(props.type)) {
+      throw new Error("The provided union type cannot be represented as an enum");
+    }
+    type = $.enum.createFromUnion(props.type);
+  } else {
+    type = props.type;
+  }
+
+  if (!props.type.name || props.type.name === "") {
+    reportDiagnostic($.program, { code: "type-declaration-missing-name", target: props.type });
+  }
+  const refkeys = declarationRefkeys(props.refkey, props.type);
+  const name = props.name ?? py.usePythonNamePolicy().getName(props.type.name!, "enum");
+  const members = Array.from(type.members.entries());
+  const doc = props.doc ?? $.type.getDoc(type);
+  const enumType = determineEnumType(members, type);
+
+  return (
+    <py.EnumDeclaration doc={doc} name={name} refkey={refkeys} baseType={enumType}>
+      <For each={members} joiner={"\n"}>
+        {([key, value]) => {
+          const memberDoc = $.type.getDoc(value);
+          return (
+            <EnumMember
+              doc={memberDoc}
+              type={value}
+              refkey={
+                $.union.is(props.type) ? efRefkey(props.type.variants.get(key)) : efRefkey(value)
+              }
+            />
```

Comment:
Does the `EnumMember` need some sort of name? Or is that being handled by the refkey?